lms/tests/test_solutions.py

```diff
@@ -184,3 +185,22 @@ def test_start_checking(
                 assert is_checking
                 the_solution = Solution.get_by_id(solution.id)
                 assert the_solution.state == Solution.STATES.IN_CHECKING.name
+
+    @staticmethod
+    def test_share_solution(
```

Comment:
Be sure to test Access Denied. Privileges must be thoroughly checked in this feature.